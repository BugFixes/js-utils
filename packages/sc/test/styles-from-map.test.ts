import {
  selectFromProps,
  selectFromTheme,
  stylesFromMap,
} from '../src'

describe('stylesFromMap()', function () {
  const props = {
    theme: {
      bgColor: '#ffffff',
      margin: null,
      textColor: '#00cc00',
    },
    zIndex: 10,
  }

  test('creates a style mapper', function () {
    const output = stylesFromMap({
      foo: 'bar',
    })

    expect(typeof output).toBe('function')
    expect(output).toHaveLength(1)
  })

  describe('generated style mapper', function () {
    test('correctly maps explicit values', function () {
      const output = stylesFromMap({
        'background-color': '#ccffff',
        'color': '#0000cc',
        'margin': null,
        'padding': undefined,
        'z-index': 12,
      })

      expect(output(props)).toMatchSnapshot()
    })

    test('correctly maps selected values', function () {
      const output = stylesFromMap({
        'background-color': selectFromTheme('bgColor'),
        'color': selectFromTheme('textColor'),
        'margin': selectFromTheme('margin'),
        'padding': selectFromTheme('padding'),
        'z-index': selectFromProps('zIndex'),
      })

      expect(output(props)).toMatchSnapshot()
    })
  })
})
