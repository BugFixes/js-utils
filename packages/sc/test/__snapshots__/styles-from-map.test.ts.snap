// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stylesFromMap() generated style mapper correctly maps explicit values 1`] = `"background-color:#ccffff;color:#0000cc;z-index:12;"`;

exports[`stylesFromMap() generated style mapper correctly maps selected values 1`] = `"background-color:#ffffff;color:#00cc00;z-index:10;"`;
